spec/tom_queue/deferred_work/deferred_work_manager_integration_spec.rb

```diff
@@ -2,6 +2,48 @@
 
 # "Integration" For lack of a better word, trying to simulate various failures
 
+describe "DeferredWorkManager", "#stop" do
+  class ExceptionReporter
+    attr_reader :filepath
+
+    def initialize(filepath)
+      @filepath = filepath
+    end
+
+    def notify(exception)
+      File.open(filepath, "w+") { |file| file.write(exception) }
```

Comment:
> Why use a file?

Because it's a forked processes, so memory isn't shared back with the original test process. Using a file means we can check cross-process easily.